auth/composite_authenticator.go

```diff
@@ -1,7 +1,13 @@
 package auth
 
+import "fmt"
+
 func NewCompositeAuthenticator(authenticators ...Authenticator) AuthenticatorFunc {
 	return AuthenticatorFunc(func(user, pass string) (bool, error) {
+		if user == "" || pass == "" {
+			return false, fmt.Errorf("username and/or password is empty")
```

Comment:
seems like a fine approach in theory, but we're just hiding the issue now. As it current stands, we're at least getting errors within auth0 itself. 

The other issue is that now we're returning all these errors to the requester, which I don't believe will mitigate this services' unresponsiveness.

Reply:
true we don't know this but I think this is still a good intermediate step. the code path to validate empty basic auth credentials and break out if empty, should be able handle hundreds of requests per second.

a simple test with d.ims.io running locally shows the following results

**with validation logic**
```
➜  ~ echo "GET http://@localhost:8181/d.ims.io/repository/channels/thechunnel/image" | vegeta attack -duration=5s | vegeta report
Requests      [total, rate]            250, 50.20
Duration      [total, attack, wait]    4.981425352s, 4.979999s, 1.426352ms
Latencies     [mean, 50, 95, 99, max]  1.167072ms, 1.310575ms, 1.790571ms, 1.822917ms, 4.981541ms
Bytes In      [total, mean]            8500, 34.00
Bytes Out     [total, mean]            0, 0.00
Success       [ratio]                  0.00%
Status Codes  [code:count]             500:250  
Error Set:
500 Internal Server Error
```

**without validation logic**
```
➜  ~ echo "GET http://@localhost:8181/d.ims.io/repository/channels/thechunnel/image" | vegeta attack -duration=5s | vegeta report
Requests      [total, rate]            250, 50.20
Duration      [total, attack, wait]    51.399746783s, 4.979999s, 46.419747783s
Latencies     [mean, 50, 95, 99, max]  24.100478605s, 30.001516521s, 38.677411144s, 44.282700149s, 46.459747783s
Bytes In      [total, mean]            8460, 33.84
Bytes Out     [total, mean]            0, 0.00
Success       [ratio]                  0.00%
Status Codes  [code:count]             0:160  500:90 
```